src/main/java/dk/cngroup/trainings/spring/springassignment/service/AnimalServiceImpl.java

```diff
@@ -86,22 +94,21 @@ public boolean deleteAnimalById(long id) {
 	}
 
 	@Override
-	public String addExistingCareTakerToExistingAnimal(long animalId, long careTakerId) {
-		Optional<Animal> animal = animalRepository.findById(animalId);
+	public void addExistingCareTakerToExistingAnimal(long animalId, long careTakerId)
+			throws AnimalNotFoundException, CareTakerNotFoundException, AnimalAndCareTakerAlreadyLinked {
+		Optional<Animal> animal = this.getAnimalById(animalId);
 		Optional<CareTaker> careTaker = careTakerRepository.findById(careTakerId);
-		if (!animal.isPresent()) {
-			return "Failed: Animal id " + animalId + " doesn't exist";
-		} else if (!careTaker.isPresent()) {
-			return "Failed: CareTaker id " + careTakerId + " doesn't exist";
+		// Animal service can't use CareTakerService because we will have recursive dependencies
```

Comment:
I think you can have circular dependencies, if you use only constructor injection in Spring. But I am not totally sure about that.

Reply:
This solution is also fine, also the comment is reasonable here

Reply:
I tried before I notices the circular dependencies and simply it gave a error when running the project.
The circular dependency in constructor injection should include some if else test to get out of the infinite circular dependency I guess.

Reply:
Some reading: http://www.baeldung.com/circular-dependencies-in-spring

But using the repository directly, when it is just CRUD operations, is absolutely fine